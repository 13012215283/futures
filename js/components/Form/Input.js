import React from 'react';
import PropTypes from 'prop-types';

import { Text, TextInput, View, StyleSheet } from 'react-native';
import { px2dp } from '../../utils/px2dp';
import noop from '../../utils/noop';

import { colors, sizes } from '../../components/themes';

const Input = ({
  label,
  placeholder,
  val,
  children,
  onChangeText,
  ...props
}) => (
  <View style={styles.containerStyle}>
    <View style={[{ justifyContent: 'center' }, styles.labelWrapperStyle]}>
      <Text style={[styles.labelText, { justifyContent: 'space-between' }]}>
        {label}
      </Text>
    </View>
    <View style={[{ flex: 1, justifyContent: 'center' }]}>
      <TextInput
        style={{ fontSize: sizes.f2 }}
        underlineColorAndroid="transparent"
        placeholder={placeholder}
        placeholderTextColor={colors[1102]}
        onChangeText={onChangeText}
        {...props}
      />
    </View>
    {React.Children.map(children, child => (
      <View style={styles.center}>{child}</View>
    ))}
  </View>
);

Input.propTypes = {
  label: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  val: PropTypes.string,
  children: PropTypes.node,
  onChangeText: PropTypes.func,
};

Input.defaultProps = {
  children: null,
  val: '',
  onChangeText: noop,
};

export default Input;

const styles = StyleSheet.create({
  center: {
    alignItems: 'flex-end',
    justifyContent: 'center',
  },
  containerStyle: {
    flexDirection: 'row',
    borderBottomWidth: 1,
    borderBottomColor: colors[1104],
    height: px2dp(104),
  },
  labelWrapperStyle: {
    width: px2dp(88),
  },
  labelText: {
    color: colors[1101],
    fontSize: sizes.f2,
    textAlign: 'justify',
  },
});
