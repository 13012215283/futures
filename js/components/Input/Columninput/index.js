import React, { Component } from 'react';
import { View, TextInput, DeviceEventEmitter } from 'react-native';
import PropTypes from 'prop-types';

import styles from './style';
import { colors } from '../../themes';

export default class Columninput extends Component {
  static defaultProps = {
    placeholder: '请输入值',
  };

  constructor(props) {
    super(props);

    this.state = {
      text: '',
    };

    this.changeText = this.changeText.bind(this);
    this.getValue = this.getValue.bind(this);
    this.onChange = this.onChange.bind(this);
  }

  onChange = () => {
    DeviceEventEmitter.emit('inputTextIsNull');
  };

  getValue = () => this.state.text;

  changeText(text) {
    this.setState({ text });
  }

  render() {
    const { placeholder } = this.props;

    return (
      <View style={styles.inputRectangle}>
        <TextInput
          style={styles.inputValue}
          underlineColorAndroid="transparent"
          placeholder={placeholder}
          placeholderTextColor={colors['1103']}
          onChangeText={this.changeText}
          onChange={this.onChange}
        />
      </View>
    );
  }
}

Columninput.propTypes = {
  placeholder: PropTypes.string,
};
