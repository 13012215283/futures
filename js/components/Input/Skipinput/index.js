import React, { Component } from 'react';
import { Text, View, TouchableWithoutFeedback } from 'react-native';
import PropTypes from 'prop-types';

import noop from '../../../utils/noop';
import styles from './style';

export default class Skipinput extends Component {
  static defaultProps = {
    title: '',
    skipTo: noop,
    value: '',
  };

  constructor(props) {
    super(props);

    this.skipTo = this.skipTo.bind(this);
  }

  skipTo() {
    this.props.skipTo();
  }

  render() {
    const { value, title } = this.props;
    return (
      <TouchableWithoutFeedback onPress={this.skipTo}>
        <View style={styles.inputRectangle}>
          <View style={[styles.inputContent, styles.inputTitle]}>
            <Text
              style={
                value === '' ? styles.inputTileText : styles.inputValueText
              }
            >
              {value || title}
            </Text>
          </View>
          <View style={[styles.inputContent, styles.inputRight]}>
            <Text style={styles.inputRightText}>&gt;</Text>
          </View>
        </View>
      </TouchableWithoutFeedback>
    );
  }
}

Skipinput.propTypes = {
  skipTo: PropTypes.func,
  title: PropTypes.string,
  value: PropTypes.string,
};
