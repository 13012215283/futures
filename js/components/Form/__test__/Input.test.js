import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Input from '../Input';

describe('<Input />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Input label="账号" placeholder="请输入姓名" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('simulates onChangeText events', () => {
    let val = '';
    const wrapper = shallow(
      <Input
        label="密码"
        placeholder="请输入密码"
        onChangeText={text => {
          val = text;
        }}
        secureTextEntry
      />
    );
    wrapper.find('TextInput').simulate('ChangeText', '888');
    expect(val).toBe('888');
  });
});
