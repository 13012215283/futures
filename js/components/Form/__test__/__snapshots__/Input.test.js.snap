// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> renders correctly 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#e0e0e0",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "height": 104,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "justifyContent": "center",
        },
        Object {
          "width": 88,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#212121",
            "fontSize": 28,
            "textAlign": "justify",
          },
          Object {
            "justifyContent": "space-between",
          },
        ]
      }
    >
      账号
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="请输入姓名"
      placeholderTextColor="#757575"
      style={
        Object {
          "fontSize": 28,
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;
